include/deal.II/numerics/vector_repartitioner.h

```diff
@@ -0,0 +1,100 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2020 by the deal.II authors
+//
+// This file is part of the deal.II library.
+//
+// The deal.II library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.md at
+// the top level directory of deal.II.
+//
+// ---------------------------------------------------------------------
+
+#ifndef dealii_vector_repartitioner_h
+#define dealii_vector_repartitioner_h
+
+
+#include <deal.II/base/config.h>
+
+#include <deal.II/base/partitioner.h>
+
+#include <deal.II/dofs/dof_handler.h>
+
+#include <deal.II/lac/la_parallel_vector.h>
+
+
+DEAL_II_NAMESPACE_OPEN
+
+/**
+ * Class for repartitioning data living on the same triangulation, which
+ * has been partitioned differently.
+ */
+template <typename VectorType>
+class VectorRepartitioner
```

Comment:
I agree that we could find some better names for the class and the operations. `VectorRepartitioner` would intuitively imply that the class takes a vector and modifies it. Instead, however, the class provides functionality to translate from one parallel partition to another one...